src/Compilers/CSharp/Portable/Lowering/BoundTreeToDifferentEnclosingContextRewriter.cs

```diff
@@ -131,11 +131,27 @@ public override BoundNode VisitAwaitableInfo(BoundAwaitableInfo node)
             var getAwaiter = (BoundExpression?)this.Visit(node.GetAwaiter);
             var isCompleted = VisitPropertySymbol(node.IsCompleted);
             var getResult = VisitMethodSymbol(node.GetResult);
-            var runtimeAsyncAwaitMethod = VisitMethodSymbol(node.RuntimeAsyncAwaitMethod);
 
             _placeholderMap.Remove(awaitablePlaceholder);
 
-            return node.Update(rewrittenPlaceholder, node.IsDynamic, getAwaiter, isCompleted, getResult, runtimeAsyncAwaitMethod);
+            BoundCall? runtimeAsyncAwaitCall = null;
+            var runtimeAsyncAwaitCallPlaceholder = node.RuntimeAsyncAwaitCallPlaceholder;
+            var rewrittenRuntimeAsyncAwaitCallPlaceholder = runtimeAsyncAwaitCallPlaceholder;
+            if (rewrittenRuntimeAsyncAwaitCallPlaceholder is not null)
+            {
+                rewrittenRuntimeAsyncAwaitCallPlaceholder = runtimeAsyncAwaitCallPlaceholder!.Update(VisitType(rewrittenRuntimeAsyncAwaitCallPlaceholder.Type));
```

Comment:
nit: the mixing of `runtimeAsyncAwaitCallPlaceholder` and `rewrittenRuntimeAsyncAwaitCallPlaceholder` is a bit strange. Consider just using `runtimeAsyncAwaitCallPlaceholder`: `runtimeAsyncAwaitCallPlaceholder!.Update(VisitType(runtimeAsyncAwaitCallPlaceholder.Type));`

Reply:
nit: Not directly related to this PR, seems like we could tighten the type of `_placeholderMap` to main purpose clearer, as it's not a map from placeholder to any sort of expression, but rather from placeholder to rewritten placeholder.